### ISSUE TITLE
"illegal generic type for instanceof" since 1.10.6

### ISSUE THREAD

**Issue by aabdagic** [google/FreeBuilder#229]
Happens in the mergeFrom(Foo value) method when the value type with generics. Here is the generated code:

```java
    if (value instanceof Foo_Builder.Value<R> && resources == ImmutableList.<R>of()) {
      resources = value.getResources();
    } else {
      addAllResources(value.getResources());
    }
```

I suppose this code here generates it:
https://github.com/google/FreeBuilder/blob/28591b548edba0ac280afb610b08b13ebc011f9f/src/main/java/org/inferred/freebuilder/processor/ListPropertyFactory.java#L407

And analogous for Sets.

**Comment by chrisalice** [comment]
Thanks for the bug report! Could you see if this is fixed in [v1.12.1-rc1](https://github.com/google/FreeBuilder/releases/tag/v1.12.1-rc1), please?

**Comment by aabdagic** [comment]
Doesn't seem like it is.

**Comment by chrisalice** [comment]
Can you give me an example to produce the bug, please?

**Comment by aabdagic** [comment]
Here it is:

```java
@FreeBuilder
public abstract class Page<R> {

  public abstract ImmutableList<R> getResources();

  public static class Builder<R> extends Page_Builder<R> {
  }
}
```

**Comment by chrisalice** [comment]
Could you try [v1.12.2-rc1](https://github.com/google/FreeBuilder/releases/tag/v1.12.2-rc1) for me?

**Comment by aabdagic** [comment]
With v1.12.2-rc1 it compiles. Thanks.

**Comment by chrisalice** [comment]
Fix released as [v1.12.2](https://github.com/google/FreeBuilder/releases/tag/v1.12.2).
